lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -393,6 +397,17 @@ TEST_F(AMFAppProcedureTest, TestRegistrationAuthSecurityCapabilityMismatch) {
   EXPECT_TRUE(ue_context_p == nullptr);
 }
 
+TEST_F(AMFAppProcedureTest, TestRegistrationPlmnMismatch) {
+  amf_ue_ngap_id_t ue_id = 0;
+  /* Send the initial UE message */
+  imsi64_t imsi64 = 0;
+  imsi64 = send_initial_ue_message_no_tmsi(
+      amf_app_desc_p, 36, 1, 1, 0, plmn,
+      initial_ue_plmn_mismatch_message_hexbuf,
+      sizeof(initial_ue_plmn_mismatch_message_hexbuf));
+  EXPECT_EQ(get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id), 0);
+}
```

Comment:
Can you add a check to validate if amf_proc_registration_reject() is hit?

Reply:
Added check with expected IDs to check the flow.